src/inspector_agent.cc

```diff
@@ -207,6 +207,7 @@ class AgentImpl {
   bool wait_;
   bool connected_;
   bool shutting_down_;
+  bool startup_failed_;
```

Comment:
This is starting to look like a state machine spread across several booleans.  Consolidating them into an enum is cleaner and easier to reason about, IMO.